services/catalog/src/simcore_service_catalog/utils/requests_decorators.py

```diff
@@ -0,0 +1,48 @@
+import asyncio
+import logging
+from asyncio import CancelledError
+from contextlib import suppress
+from functools import wraps
+from typing import Any, Callable, Coroutine
+
+from fastapi import Request, Response
+
+logger = logging.getLogger(__name__)
+
+_DEFAULT_CHECK_INTERVAL_S: float = 0.5
+
+
+async def _cancel_task_if_client_disconnected(
+    request: Request, task: asyncio.Task, interval: float = _DEFAULT_CHECK_INTERVAL_S
+) -> bool:
+    with suppress(CancelledError):
+        while True:
```

Comment:
what happens if the task finishes without being canceled? Will `request.is_disconnected()` always be called at some point?